faith_presentation/src/commonMain/kotlin/net/thechance/mena/faith/presentation/feature/quran/tilwah/TilawahViewModel.kt

```diff
@@ -54,8 +103,14 @@ class TilawahViewModel(val quranRepository: QuranRepository) :
     }
 
     private fun getAllRecitersSuccessfully(reciters: List<Reciter>) {
-        val recitersUi = reciters.map {
-            ReciterUi(
+        val filteredReciters = if (tilawahArgs.surahId != null) {
+            filterRecitersForSurah(reciters, tilawahArgs.surahId)
+        } else {
+            reciters
+        }
```

Comment:
```suggestion
        val filteredReciters =
         if (tilawahArgs.surahId != null) filterRecitersForSurah(reciters, tilawahArgs.surahId)
         else reciters
```
add i support @Mostafa-alsaygh opinion for single respoinsibility